test-app/app/models/blog.rb

```diff
@@ -6,7 +6,13 @@ class Blog < ActiveRecord::Base
   alias_attribute :headline, :title
   alias_attribute :text, :content
 
-  enum status: [:draft, :published, :archived]
+  # Rails 7.1 deprecated defining enums with keyword arguments.
+  # See https://github.com/rails/rails/pull/50987
+  if Gem::Version.new(Rails.version) >= Gem::Version.new("7.1")
```

Comment:
according to 
- https://api.rubyonrails.org/v6.1.0/classes/ActiveRecord/Enum.html
- https://api.rubyonrails.org/v7.0.0/classes/ActiveRecord/Enum.html

The new syntax is the suggested one starting from 7.0.0, so the check can change to 

```suggestion
  if Gem::Version.new(Rails.version) >= Gem::Version.new("7.0")
```

(even if this will leave out beta and rc versions)

Reply:
```suggestion
  if Gem::Requirement.new(">= 7.0.pre") =~ Gem::Version.new(Rails.version)
```

The `Gem::Requirement.new(">= 7.0.pre")` condition will correctly include all the versions listed from :
- 7.0.0 and up, including any pre-release versions (like beta or rc) for any of the major versions (7.0, 7.1, 7.2).

`7.0.pre`:
The .pre suffix in the requirement includes all pre-release versions of 7.0 and above, such as alpha, beta, and rc versions. It also includes the final release version 7.0.0 and any subsequent versions.

This condition will match any version that is 7.0.0 or higher, including 7.0.0.alpha, 7.0.0.beta, 7.0.0.rc, and any stable versions like 7.0.1, 7.1.0, 7.2.0, etc.